back-end/hub-core/src/main/java/io/apicurio/hub/core/storage/jdbc/CommonSqlStatements.java

```diff
@@ -138,28 +142,46 @@ public String deleteLinkedAccounts() {
     public String insertApiDesign() {
         return "INSERT INTO api_designs (name, description, created_by, created_on, tags, api_type) VALUES (?, ?, ?, ?, ?, ?)";
     }
-    
+
     /**
      * @see io.apicurio.hub.core.storage.jdbc.ISqlStatements#deleteApiDesign()
      */
     @Override
     public String deleteApiDesign() {
         return "DELETE FROM api_designs WHERE id = ?";
     }
-    
+
     /**
      * @see io.apicurio.hub.core.storage.jdbc.ISqlStatements#selectApiDesigns()
      */
     @Override
     public String selectApiDesigns() {
-    	if (shareForEveryone) {
-    		return "SELECT d.* FROM api_designs d";
-    	}
+        if (shareForEveryone) {
+            return "SELECT d.* FROM api_designs d";
+        }
         return "SELECT d.* FROM api_designs d JOIN acl a ON a.design_id = d.id WHERE a.user_id = ?";
     }
-    
+
     @Override
     public String selectRecentApiDesigns() {
+        // If datasource is Azure SQL
+        if(config.getJdbcType().equals("azuresql")) {
+            if (shareForEveryone) {
+                return "SELECT TOP 5 c.design_id, MAX(c.version) AS version "
+                        + "FROM api_content c "
+                        + "WHERE c.created_by = ? "
+                        + "GROUP BY c.design_id "
+                        + "ORDER BY version DESC ";
+            } else {
+                return "SELECT TOP 5 c.design_id, MAX(c.version) AS version "
+                        + "FROM api_content c "
+                        + "JOIN acl a ON a.design_id = c.design_id "
+                        + "WHERE a.user_id = ? AND c.created_by = ? "
+                        + "GROUP BY c.design_id "
+                        + "ORDER BY version DESC ";
+            }
+        }
+
```

Comment:
For any AzureSQL specific SQL statements, you should simply override the appropriate method in your new `AzureSQLStatements` class rather than have an `if` statement here.  Could you move all the appropriate azure-specific sql statement changes into that class?